packages/story-editor/src/utils/storyUpdates.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import cleanForSlug from './cleanForSlug';
+
+/**
+ * For use on blur of updating story title
+ * that can impact what story slug is if
+ * not set independently.
+ *
+ * @param {Object} args Necessary data for update.
+ * @param {string} args.currentSlug The currently assigned value to story's slug.
+ * @param {string} args.currentTitle The currently assigned value to story's title.
+ * @param {number} args.storyId The currently assigned value to story's id.
+ * @param {Function} args.updateStory The callback to useStory's action for updating the story with new slug.
+ * @return {void}
+ */
+export const updateSlug = ({
+  currentSlug,
+  currentTitle,
+  storyId,
+  updateStory,
+}) => {
+  if (!currentSlug || parseInt(currentSlug) === storyId) {
```

Comment:
Perhaps the check without `parseInt` was for when the story was still new and hadn't been saved at all, and then changing the slug, could it be? Or perhaps some non-standard permalink setting?
I'll double check tomorrow if that's not the case (finished the day already).

Reply:
If the story is new and hasn't been saved at all then the slug's empty. 

The story id is used as a fallback if there's no story title. So, for instance, if I focus the story title and then tab out of it, there's still no viable story title in which to make a slug out of so the story id is used as a fallback. 

Where the parseInt comes into play is if the current slug is equal to the story id on a story title update then we should check and see if there's an update, otherwise the slug isn't updated every time the story title changes (which is consistent with staging). I honestly think this shoulda been a parseInt all along but it's such an edge case it wasn't really noticed. I did set up unit tests now so we can be confident in when this is updating and how.

Reply:
Ah, okay, got it. The only place where I can see this could now cause issues is if the slug starts with the story ID.
E.g. the story ID is 3 and the slug is `3-things-you-can-do`, then blurring the title would overwrite that slug, if I understand correctly.

Reply:
`Number()` seems preferable in that case over `parseInt`